src/util/modules.py

```diff
@@ -51,9 +51,28 @@
 from .records import ref, remove_menus, remove_records, remove_view, replace_record_references_batch
 
 INSTALLED_MODULE_STATES = ("installed", "to install", "to upgrade")
-NO_AUTOINSTALL = str2bool(os.getenv("UPG_NO_AUTOINSTALL", "0")) if version_gte("15.0") else False
 _logger = logging.getLogger(__name__)
 
+if version_gte("15.0"):
+    AUTO_INSTALL = os.getenv("UPG_AUTOINSTALL")
+    _NO_AUTOINSTALL = os.getenv("UPG_NO_AUTOINSTALL")
+    if AUTO_INSTALL:
+        if _NO_AUTOINSTALL:
+            _logger.info("Ignoring the deprecated `UPG_NO_AUTOINSTALL` env variable as `UPG_AUTOINSTALL` is also set.")
+    elif _NO_AUTOINSTALL:
+        # TODO add link to doc
+        _logger.warning(
```

Comment:
I'd move this to after setting AUTO_INSTALL and add the AUTO_INSTALL value we got to the log line, plus the original UPG_NO_AUTOINSTALL value.